Funcky.Analyzer/Funcky.Analyzer/EnumerableRepeatNeverAnalyzer.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using static Funcky.Analyzer.LocalizedResourceLoader;
+using static Funcky.Analyzer.Resources;
+
+namespace Funcky.Analyzer
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class EnumerableRepeatNeverAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = nameof(EnumerableRepeatNeverAnalyzer);
+        private const string Category = nameof(Funcky);
+
+        private static readonly LocalizableString Title = LoadFromResource(nameof(EnumerableRepeatNeverAnalyzerTitle));
+        private static readonly LocalizableString MessageFormat = LoadFromResource(nameof(EnumerableRepeatNeverAnalyzerMessageFormat));
+        private static readonly LocalizableString Description = LoadFromResource(nameof(EnumerableRepeatNeverAnalyzerDescription));
+
+        private static readonly DiagnosticDescriptor Rule = new(nameof(EnumerableRepeatNeverAnalyzer), Title, MessageFormat, Category, DiagnosticSeverity.Warning, isEnabledByDefault: true, description: Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+
+            context.RegisterSyntaxNodeAction(FindEnumerableRepeateNever, SyntaxKind.InvocationExpression);
+        }
+
+        private void FindEnumerableRepeateNever(SyntaxNodeAnalysisContext context)
+        {
+            if (IsRepeatNever(new SyntaxMatcher(context)))
+            {
+                context.ReportDiagnostic(CreateDiagnostic(context));
+            }
+        }
+
+        private static bool IsRepeatNever(SyntaxMatcher syntax)
+            => syntax.MatchStaticCall(nameof(Enumerable), nameof(Enumerable.Repeat))
+                && syntax.MatchArgument(Argument.Second, 0);
```

Comment:
you could even detect the use of constants if you would use the `IOperation` API.
See: [`ConstantValue`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.ioperation.constantvalue?view=roslyn-dotnet-3.10.0#Microsoft_CodeAnalysis_IOperation_ConstantValue)

Reply:
context.[SemanticModel](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.diagnostics.syntaxnodeanalysiscontext.semanticmodel?view=roslyn-dotnet-3.10.0).[GetOperation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.semanticmodel.getoperation?view=roslyn-dotnet-3.10.0#Microsoft_CodeAnalysis_SemanticModel_GetOperation_Microsoft_CodeAnalysis_SyntaxNode_System_Threading_CancellationToken_)(context.Node, context.CancellationToken)